website/blog/2021-02-04-raptorx.md

```diff
@@ -0,0 +1,214 @@
+---
+title: RaptorX: Building a 10X Faster Presto
+author: James Sun
+authorURL: https://www.linkedin.com/in/yutiansun/
+authorFBID: 100001087056694
+---
+<div style="text-align: justify">
+
+
+**Facebook:** Amit Dutta, Baldeep Hira, Biswapesh Chattopadhyay, James Sun, Jialiang Tan, Ke Wang, Lin Liu, Naveen Cherukuri, Nikhil Collooru, Peter Na, Rohit Jain, Saksham Sachdev, Sergey Pershin, Shixuan Fan, Varun Gajjala
+
+**Alluxio:** Bin Fan, Calvin Jia, Haoyuan Li
+
+**Twitter:** Zhenxiao Luo
+
+**Pinterest:** Lu Niu
+
+*RaptorX is an internal project name aiming to boost query latency significantly beyond what vanilla Presto is capable of. This blog post introduces the hierarchical cache work, which is the key building block for RaptorX. With the support of the cache, we are able to boost query performance by 10X. This new architecture can beat performance oriented connectors like Raptor with the added benefit of continuing to work with disaggregated storage.*
+
+
+
+<!--truncate-->
+
+
+## Problems with Disaggregated Storage
+
+[Disaggregated storage](https://en.wikipedia.org/wiki/Disaggregated_storage) is the industry wide trend towards scaling storage and compute independently. It helps cloud providers to save costs. Presto by nature supports such architecture. Data can be streamed from remote storage nodes outside of Presto servers.
+
+
+
+However, storage-compute disaggregation also provides new challenges for query latency as scanning huge amounts of data over the wire is going to be IO bound if when the network is saturated. Moreover, the metadata paths will also go through the wire to retrieve the location of the data; a few roundtrips of metadata RPCs can easily bump up the latency to more than a second. The following figure shows the IO paths for Hive connectors in orange color, each of which could be a bottleneck on query performance.
+
+
+
+![Presto Architecture](/img/blog/2021-02-04-raptorx/presto-arch.png)
+
+
+
+
+## RaptorX: Build a Hierarchical Caching Solution
+
+Historically, to solve the network saturation issue, Presto has a built-in Raptor connector to load data from remote storage to local SSD for fast access. However, this solution is no different from having a shared compute/storage node, which is against the idea of storage-compute disaggregation. The downside is obvious: either we waste CPU because the SSD in a worker is full or wasting SSD capacity if we are CPU bound. Thus, we started project RaptorX.
```

Comment:
```suggestion
Historically, to solve the network saturation issue, Presto has a built-in Raptor connector to load data from remote storage to local SSD for fast access. However, this solution is no different from having a shared compute/storage node, which is runs counter to the idea of storage-compute disaggregation. The downside is obvious: either we waste CPU because the SSD in a worker is full or we waste SSD capacity if we are CPU bound. Thus, we started project RaptorX.
```